### ISSUE TITLE
optionalfields linter does not detect enum marker on type

### ISSUE THREAD

**Issue by sbueringer** [kubernetes-sigs/kube-api-linter#113]
In the following case the optionalfields linter does not detect that a field has enum validation. Because of that the linter assume that the empty value is valid.

```
// Format specifies the output format of the bootstrap data
// +kubebuilder:validation:Enum=cloud-config;ignition
type Format string

const (
	// CloudConfig make the bootstrap data to be of cloud-config format.
	CloudConfig Format = "cloud-config"

	// Ignition make the bootstrap data to be of Ignition format.
	Ignition Format = "ignition"
)

...

	// format specifies the output format of the bootstrap data
	// +optional
	Format Format `json:"format,omitempty"`
```

Finding:
```
api/bootstrap/kubeadm/v1beta2/kubeadmconfig_types.go:118:2: optionalfields: field Format is optional and has a valid zero value (""), but the validation is not complete (e.g. minimum length). The field should be a pointer to allow the zero value to be set. If the zero value is not a valid use case, complete the validation and remove the pointer. (kubeapilinter)
        Format Format `json:"format,omitempty"`
        ^
```
